app/src/main/java/org/jboss/hal/client/accesscontrol/RoleColumn.java

```diff
@@ -362,6 +363,31 @@ private void editStandardRole(Role role) {
                             });
                 }).show(modelNode);
         form.getFormItem(NAME).setValue(role.getName());
+*/
+
+        // Promise code:
+        form.getFormItem(NAME).setValue(role.getName());
+        new ModifyResourceDialog(resources.messages().modifyResourceTitle(resources.constants().role()), form)
+                .showWithPromise(modelNode).then(changedValues -> {
```

Comment:
This is an example usage of the promise-based variant in `ModifyResourceDialog`.